packages/react-events/src/__tests__/Press-test.internal.js

```diff
@@ -2301,4 +2303,58 @@ describe('Event responder: Press', () => {
       },
     ]);
   });
+
+  it('should properly only flush sync once when the event systems are mixed', () => {
+    const ref = React.createRef();
+    let renderCounts = 0;
+
+    function dispatchEvent(name) {
+      const event = createEvent(name);
+      Object.defineProperty(event, 'timeStamp', {
+        value: 100,
+      });
+      window.event = event;
+      ref.current.dispatchEvent(event);
+      window.event = null;
+    }
+
+    function MyComponent() {
+      const [, updateCounter] = React.useState(0);
+      renderCounts++;
+
+      function handlePress() {
+        updateCounter(count => count + 1);
+      }
+
+      return (
+        <div>
+          <Press onPress={handlePress}>
+            <button
+              ref={ref}
+              onClick={() => {
+                updateCounter(count => count + 1);
+              }}>
+              Press me
+            </button>
+          </Press>
+        </div>
+      );
+    }
+
+    const newContainer = document.createElement('div');
+    const root = ReactDOM.unstable_createRoot(newContainer);
+    document.body.appendChild(newContainer);
+    root.render(<MyComponent />);
+    Scheduler.flushAll();
+
+    dispatchEvent('pointerdown');
+    dispatchEvent('pointerup');
+    dispatchEvent('click');
+
+    expect(renderCounts).toBe(2);
+    Scheduler.flushAll();
+    expect(renderCounts).toBe(4);
+
+    document.body.removeChild(newContainer);
+  });
```

Comment:
Can you add a test that includes a subsequent event sequence? To check that `interactiveEventTimeStamp` gets reset properly.

Reply:
Added some further tests to verify this works.